component/block_cache/block_cache.go

```diff
@@ -1598,11 +1598,31 @@ func (bc *BlockCache) commitBlocks(handle *handlemap.Handle) error {
 		listMap[k].committed = true
 	}
 
+	if restageId != "" {
```

Comment:
There can be multiple occurrences in a file where the block staged had size less than the configured block size

Reply:
Have taken care of this as well now, kindly review.